dev_roadmap.md

```diff
@@ -0,0 +1,329 @@
+
+------------------------------------
+
+## Library package structure
+
+With a name as general as Quantum Toolkit in Python, the scope for new code modules to be added to qutip is very wide. The library was becoming increasingly difficult to maintain, and in c. 2020 the QuTiP Admim Team decided to limit the scope of the 'main' (for want of a better name) qutip package. 
+This scope is restricted to components for the simulation (solving) of the dynamics of quantum systems.
+The scope includes utilities to support this, including analysis and visualisation of output.
+
+At the same time, again with the intention of easing maintence, a decision to limit dependences was agreed upon.
+Main qutip runtime code components should depend only upon Numpy and Scipy. 
+Installation (from source) requires Cython, and some optional components also require Cython at runtime.
+Unit testing requires Pytest. Visualisation (optional) components require Matplotlib.
+
+Due to the all encompassing nature of the plan to abstract the linear algebra data layer, this enhancement (developed as part of a GSoC project) was allowed the freedom (potential for non-backward compatibility) of requiring a major release. The timing of such allows for a restructuring of the qutip compoments, such that some that could be deemed out of scope could be packaged in a different way -- that is, not installed as part of the main qutip package. Hence the proposal for different types of package described next. With reference to the [discussion above](#what-is-qutip) on the name QuTiP/qutip, the planned restructuring suffers from confusing naming, which seems unavoidable without remaining either the organisation or the main package (neither of which are desirable).
+
+ - QuTiP family packages. The main qutip package already has sub-packages, which are maintained in the main qutip repo. Any packages maitained by the QuTiP organisation will be called QuTiP 'family' packages. Sub-packages within qutip main will be called 'integrated' sub-packages. Some packages will be maintained in their own repos and installed separately within the main qutip folder structure to provide backwards compatibility, these are (will be) called qutip optional sub-packages. Others will be installed in their own folders, but (most likely) have qutip as a dependency -- these will just be called 'family' packages.
+ - QuTiP affilliated packages. Other packages have been developed by others outside of the QuTiP organisation that work with, and are complementary to, qutip. The plan is to give some recognition to those that we deserve worthy of such [this needs clarification]. These packages will not be maintained by the QuTiP Team.
+ 
+ &#x1F537; Do we really need optional subpackages? It seems that are a bit fiddly and as we are side-stepping bw compat with a major version, we could just make QIP a separate package.
```

Comment:
> That's a fair point. The intention of making it an optional package (plugin) was indeed for backward compatibility, to make it as close as possible to the old version. The same holds for `control` and `lattice` I guess? We can surely discuss this. But if we decide to be bold and make all of them separate package, we should let Sidhant know as soon as possible since this will reduce his work.

Yes, we should discuss this very soon and let @sarsid know what we decide.